video/out/vo_gpu.c

```diff
@@ -429,13 +317,57 @@ static int preinit(struct vo *vo)
     return -1;
 }
 
-#define OPT_BASE_STRUCT struct gl_priv
+#define OPT_BASE_STRUCT struct gpu_priv
+static const m_option_t options[] = {
+    OPT_STRING_VALIDATE("gpu-context", context_name, 0,
+                        ra_ctx_validate_context),
+    OPT_CHOICE("gpu-api", context_type, 0,
+               ({"auto",  RA_API_ANY},
+               {"opengl", RA_API_OPENGL})),
+    OPT_FLAG("gpu-debug", opts.debug, 0),
+    OPT_FLAG("gpu-sw", opts.allow_sw, 0),
+    OPT_INTRANGE("swapchain-depth", opts.swchain_depth, 0, 1, 8),
+    {0}
+};
+
+static const struct gpu_priv defaults = { .opts = {
+    .swchain_depth = 3,
+}};
+
+const struct vo_driver video_out_gpu = {
+    .description = "GPU-accelerated Renderer",
+    .name = "gpu",
+    .caps = VO_CAP_ROTATE90,
+    .preinit = preinit,
+    .query_format = query_format,
+    .reconfig = reconfig,
+    .control = control,
+    .get_image = get_image,
+    .draw_frame = draw_frame,
+    .flip_page = flip_page,
+    .wait_events = wait_events,
+    .wakeup = wakeup,
+    .uninit = uninit,
+    .priv_size = sizeof(struct gpu_priv),
+    .priv_defaults = &defaults,
+    .options = options,
+};
+
+static int preinit_opengl(struct vo *vo)
+{
+    MP_WARN(vo, "--vo=opengl was replaced by --vo=gpu --gpu-api=opengl, and may"
+            " be removed in the future!\n");
+
+    struct gpu_priv *p = vo->priv;
+    p->context_type = RA_API_OPENGL;
+    return preinit(vo);
+}
 
 const struct vo_driver video_out_opengl = {
-    .description = "Extended OpenGL Renderer",
+    .description = "GPU-accelerated Renderer",
```

Comment:
That's not quite correct. We use the 3D API, instead of fixed function video functions. Both are "GPU", but this code uses 3D API only. Unless we add support for various overlay methods (for example D3D can directly render and scale video into a D3D swapchain surface and you can draw other stuff like OSD on top).

Reply:
How about “Shader-based GPU Renderer”?

Reply:
Yes, that's pretty good.

Reply:
changed